MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.App.Tests/AppTests.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using MyPerfectOnboarding.Api.Controllers;
+using MyPerfectOnboarding.Contracts.Database;
+using MyPerfectOnboarding.Contracts.Models;
+using MyPerfectOnboarding.Contracts.Services.Generators;
+using MyPerfectOnboarding.Contracts.Services.Location;
+using MyPerfectOnboarding.Dependency;
+using MyPerfectOnboarding.Dependency.Containers;
+using MyPerfectOnboarding.Tests.Utils.Builders;
+using MyPerfectOnboarding.Tests.Utils.Comparers;
+using MyPerfectOnboarding.Tests.Utils.Extensions;
+using NSubstitute;
+using NUnit.Framework;
+using Unity;
+
+namespace MyPerfectOnboarding.App.Tests
+{
+    [TestFixture]
+    internal class AppTests
+    {
+        private ListController _listController;
+        private IUrlLocator _location;
+        private ITimeGenerator _timeGenerator;
+        private IGuidGenerator _guidGenerator;
+        private IListRepository _listRepository;
+
+        [SetUp]
+        public void Init()
+        {
+            _location = Substitute.For<IUrlLocator>();
+            _timeGenerator = Substitute.For<ITimeGenerator>();
+            _guidGenerator = Substitute.For<IGuidGenerator>();
+            _listRepository = Substitute.For<IListRepository>();
+
+            var routeNames = Substitute.For<IControllersRouteNames>();
+            var connectionDetails = Substitute.For<IConnectionDetails>();
+
+            var unityContainer = new UnityContainer();
+            var container = new Container(unityContainer);
+
+            var dependencyContainerConfig = new DependencyContainerConfig(routeNames, connectionDetails);
+            dependencyContainerConfig.RegisterTypes(container);
+
+            container.Register(_location);
+            container.Register(_timeGenerator);
+            container.Register(_guidGenerator);
+            container.Register(_listRepository);
+
+            _listController = container.UnityContainer.Resolve<ListController>();
+            _listController.Request = new HttpRequestMessage();
+            _listController.Configuration = new HttpConfiguration();
+        }
+
+        [Test]
+        public async Task GetAllItemsThreeTimes()
```

Comment:
```suggestion
        public async Task GetAllItems_ThreeTimes_DatabaseOnlyReadOnce()
```
Try to stick with regular naming - `<NameOfTestedMethod>`\_`<ConditionsOrParametersDescription>`\_`<ResultsOrOutcomeDescription>` – it will tell more about the context of the text…